packages/core/admin/admin/src/content-manager/components/DynamicZone/tests/index.test.js

```diff
@@ -299,9 +299,10 @@ describe('DynamicZone', () => {
 
       const [draggedItem] = getAllByRole('button', { name: 'Drag' });
 
-      draggedItem.focus();
-
-      await user.keyboard('[Space]');
+      await waitFor(() => {
+        draggedItem.focus();
+        user.keyboard('[Space]');
+      });
```

Comment:
Can you write a comment how you cam to understand this `waitFor` being required? I don't really understand it myself at the moment. It feels like an abuse of `waitFor` considering neither function throws and you should typically await a user action 🤔

Reply:
This is something really strange at the moment. For example: I have noticed we always have to await as soon as a Select | Combobox from the design-system is involved in rendering (not even a user interaction), but I don't know what the origin of that is.

We had a quick discussion which format is appropriate for `waitFor`. The docs say:

> Returning a falsy condition is not sufficient to trigger a retry, the callback must throw an error in order to retry the condition.

> If you return a promise in the waitFor callback (either explicitly or implicitly with the async syntax), then the waitFor utility does not call your callback again until that promise rejects. This allows you to waitFor things that must be checked asynchronously.

This makes me wonder:

- what would throw in this instance, because `draggedItem` has already by queried.
- should we return `user.keyboard`, because it returns a Promise?

Reply:
I'm not sure to follow, what should we do about this? Because it might be the solution we had in mind for fixing other tests too

Reply:
Just to untangle this a bit, it works but it's not quite right. The issue is `dragged.focus` causes react updates – it shows us a tooltip, so actually i think we just need to wrap all those calls in `act`:

```ts
      act(() => {
        draggedItem.focus();
      });
```

the user event shouldn't need it because it's already wrapped internally.

Reply:
Did you apply it to all the `draggedItem.focus();` calls? I couldn't see any? If you're still getting them can you share a screenshot pls

Reply:
It works! I've only replaced one 😓